core/src/main/java/org/mapfish/print/map/style/SLDParserPlugin.java

```diff
@@ -110,6 +120,18 @@ private String removeIndexReference(final String styleString) {
         try {
             reader = charSource.openBufferedStream();
 
+            // check if the XML is valid
+            // this is only done in a separate step to avoid that fatal errors show up in the logs
+            // by setting a custom error handler.
+            DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+            dbf.setNamespaceAware(true);
+            DocumentBuilder db = dbf.newDocumentBuilder();
+            db.setErrorHandler(new ErrorHandler());
+            db.parse(new InputSource(reader));
+            Closeables.close(reader, true);
```

Comment:
There is a `Closeables.close` in the `finally` block. But I must admit it's confusing that I am reusing the variable `reader`. I will refactor this.